examples/compose/plot_compare_reduction.py

```diff
@@ -119,7 +117,8 @@
 grid.fit(digits.data, digits.target)
 
 # Delete the temporary cache before exiting
-rmtree(cachedir)
+memory.clear(warn=False)
+memory.store_backend.clear_location(location)
```

Comment:
IMO this is not a good idea. `store_backend` is not documented publicly. I would use the `rmtree` to remove the folder. In addition, I would add a comment that `memory.clear()` remove the cache used for the computation and `rmtree()` will remove the folder where the cache was localized.

Reply:
I'd rather agree with you. I add `clear_location(location)` in my second commit for removing location directory after discussions above. IMHO, it isn't necessary to remove the folder, because it can be use in the future. If we're willing to accept this, than my first commit is actual. In another case, it might be better to use `rmtree `. My original wish was to replace three libs with one.

Reply:
> IMHO, it isn't necessary to remove the folder, because it can be use in the future

In general, I would agree. However, we don't want to have a remaining folder after the example execution. That's why, I would make a comment before the `rmtree` explaining why there is this call and that in practice, you probably don't need it.

> My original wish was to replace three libs with one.

It does not matter since they come in the standard library of Python

Reply:
OK, I've made new commit.